docs/core/event_handler/api_gateway.md

```diff
@@ -214,17 +214,145 @@ Each dynamic route you set must be part of your function signature. This allows
 ???+ tip
     You can also nest dynamic paths, for example `/todos/<todo_id>/<todo_status>`.
 
+#### Routing Rules Syntax
+
+The routing system uses a specific syntax to define dynamic URL patterns. Understanding this syntax is crucial for creating flexible and robust API routes.
+
+##### Dynamic Path Parameters
+
+Dynamic path parameters are defined using angle brackets `<parameter_name>` syntax. These parameters are automatically converted to regex patterns for efficient route matching.
+
+**Syntax**: `/path/<parameter_name>`
+
+* **Parameter names** must contain only word characters (letters, numbers, underscore)
+* **Captured values** can contain letters, numbers, underscores, and these special characters: `-._~()'!*:@,;=+&$%<> \[]{}|^`
+
+=== "routing_syntax_basic.py"
+
+    ```python
+    from aws_lambda_powertools.event_handler import APIGatewayRestResolver
+
+    app = APIGatewayRestResolver()
+
+    @app.get("/users/<user_id>")
+    def get_user(user_id: str):
+        # user_id can be: "123", "user-456", "john.doe", "user_with_underscores"
+        return {"user_id": user_id}
+
+    @app.get("/orders/<order_id>/items/<item_id>")
+    def get_order_item(order_id: str, item_id: str):
+        # Multiple parameters: /orders/ORD-123/items/ITEM_456
+        return {"order_id": order_id, "item_id": item_id}
+    ```
```

Comment:
Could you create a file in the `examples/event_handler_rest/src` folder, move this example there, and refer to it here? We don't add inline examples because we want to ensure our examples are accurate and we can pass linter and formats in those files.

Also, this example is missing the Lambda entrypoint, aka `lambda handler`. please check other examples and use `LambdaContext` to annotate the context parameter.